docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2020 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+Releasing
+=========
+
+In general, you have two types of releases:
+
+- **Major/Minor-level release** - Often these releases add significant new
+  features and needs to go through extensive testing.
+- **Patch-level release** - Often these releases fix security issues or
+  important bugs.
+
+For major/minor-level releases, you'll often be working only with the
+``master``-branch and only make a single PyPI release. For patch-level
+releases you'll often be working with multiple maintenance branches and make
+several PyPI releases for all currently maintained versions of Invenio.
+
+Depending on the package that you are releasing, you will have to follow
+a different procedure. You might have to release:
+
+- one of the Python or JavaScript/React modules (happening rather often)
+- the Invenio framework, the ``invenio`` package (happening less often)
+
+Before you start
+----------------
+Prior to preparing a new Invenio release, please ensure you have the latest
+changes in your local environment for the maintenance branches and ``master``:
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
+3. Checkout maintenance or master branch:
+    - ``git checkout maint-<major>.<minor>``
+4. Merge or reset to latest changes:
+    - ``git merge --ff-only inveniosoftware/maint-<major>.<minor>``
+    - or
+    - ``git reset --hard inveniosoftware/maint-<major>.<minor>``
+
+While above is not strictly necessary, it avoids situations where you local
+and GitHub branches have diverged.
+
+Patch-level release
+-------------------
+Most of the time, patch-level releases are not necessary for the main
+``invenio`` package, since by patch-level releases of the underlying
+modules are automatically picked up: the dependencies upper bounds are
+normally allowing the installation of patch-level releases.
```

Comment:
Minor: Maybe we can add a link to illustrate this? Like [this one](https://github.com/inveniosoftware/invenio/blob/0df2bdedbf3b40ea970eed1883f465e24e26d100/setup.py#L24-L83), I am not sure since this is a permalink to current master and things may change.. what do you think?

Reply:
I think point to master is like point to line... it might change a lot in the future. If needed I would add one example. e.g.:

```python
        'invenio-records-rest>=1.7.1,<1.8.0',
        'invenio-records-ui>=1.2.0a1,<1.3.0',
```

Reply:
The link to the commit is persistent, a permalink, unless you push force and re-write the commit. I will add the example.